### DISCUSSION TITLE
screen share and webcam video on at the same time

### DISCUSSION THREAD

**Discussion by itismahmood** [mrgnr/camus#15] [Ideas]
#Feature Request
please make it possible, not to turn off webcam's video whenever we turn on screen sharing.
currently, we have only either screen shared or video, not both.
thanks

**Comment by mrgnr** [comment]
This would be a nice feature to have. Unfortunately, most of the current code assumes that each peer has a single audio and video track. I think that with some work, the code can be generalized to allow multiple audio & video tracks for each peer. The first step is to support multiple tracks in the WebRTC client (which I've started work on in this [branch](https://github.com/mrgnr/camus/compare/multivideo)). The second step is changing the [feed management](https://github.com/mrgnr/camus/blob/multivideo/camus/static/js/slices/feeds.js) in the React app. The final step would be to change the UI (in [React components](https://github.com/mrgnr/camus/tree/master/camus/static/js/components)) to display both the camera and the screenshare.
